lib/globus_client/endpoint.rb

```diff
@@ -166,6 +166,22 @@ def access_request(permissions:)
       UnexpectedResponse.call(response)
     end
 
+    def update_access_request(permissions:)
```

Comment:
If the rule is not found when updating the rule, something is probably off. We want to raise an error, not create a new rule.